src/services/highcharts-factory/highcharts-factory.svc.coffee

```diff
@@ -82,22 +87,23 @@ angular
             y: -5
         }]
 
-    addThresholds: (options = @options)->
-      return if _.isEmpty(@hc)
+    addThresholds: (chart, options = @options)->
+      return if _.isEmpty(chart)
       # Remove existing thresholds
-      _.each(@hc.series, (s)-> s.remove() if s.name.toLowerCase().includes('threshold'))
-      return @hc if _.isEmpty(options.thresholds)
+      _.each(chart.series, (s)-> s.remove() if s.name.toLowerCase().includes('threshold'))
```

Comment:
Do we have to remove all the existing threshold series? What if we want to define several thresholds on a chart in the future?